include/jemalloc/internal/psset.h

```diff
@@ -29,6 +39,12 @@ struct psset_s {
 	 */
 	edata_heap_t pageslabs[PSSET_NPSIZES];
 	bitmap_t bitmap[BITMAP_GROUPS(PSSET_NPSIZES)];
+	/*
+	 * Full slabs don't live in any edata heap.  But we still track their
+	 * stats.
+	 */
+	psset_bin_stats_t fullstats;
```

Comment:
nit: maybe `fullslab_stats`?

Shall we also track # of empty slabs, so that we know how much can be potentially purged?

Reply:
Will do.

The psset doesn't contain any empty slabs; they get evicted immediately upon becoming empty (in psset_dalloc).